### ISSUE TITLE
Failed due to progress deadline

### ISSUE THREAD

**Issue by odidev** [hashicorp/nomad-open-telemetry-getting-started#4]
Hi Team, 

I have been working on running open-telemetry in different deployment scenarios, as described in the [Getting Started](https://opentelemetry.io/docs/collector/getting-started/) guide on ubuntu amd64 platform. 

Although while deploying the Gateway and Demo, I am facing ```Failed due to progress deadline``` issue. Please have a look at the logs below: 

- **Gateway** 

```
ubuntu@ip-172-31-9-230:~/nomad-open-telemetry-getting-started$ nomad run examples/nomad/otel-collector.nomad
==> 2022-01-18T11:17:46Z: Monitoring evaluation "5d85c01f"
    2022-01-18T11:17:46Z: Evaluation triggered by job "otel-collector"
==> 2022-01-18T11:17:47Z: Monitoring evaluation "5d85c01f"
    2022-01-18T11:17:47Z: Evaluation within deployment: "49a15ed3"
    2022-01-18T11:17:47Z: Allocation "fb6ab394" created: node "cab61feb", group "otel-collector"
    2022-01-18T11:17:47Z: Evaluation status changed: "pending" -> "complete"
==> 2022-01-18T11:17:47Z: Evaluation "5d85c01f" finished with status "complete"
==> 2022-01-18T11:17:47Z: Monitoring deployment "49a15ed3"
  ! Deployment "49a15ed3" failed

    2022-01-18T11:27:46Z
    ID          = 49a15ed3
    Job ID      = otel-collector
    Job Version = 0
    Status      = failed
    Description = Failed due to progress deadline

    Deployed
    Task Group      Desired  Placed  Healthy  Unhealthy  Progress Deadline
    otel-collector  1        4       0        4          2022-01-18T11:27:46Z
```

- **Demo** 

```
ubuntu@ip-172-31-9-230:~/nomad-open-telemetry-getting-started$ nomad run examples/nomad/otel-demo.nomad
==> 2022-01-18T11:31:38Z: Monitoring evaluation "1f5b0e32"
    2022-01-18T11:31:38Z: Evaluation triggered by job "otel-demo"
    2022-01-18T11:31:38Z: Allocation "211d3a3b" created: node "cab61feb", group "load-generators"
    2022-01-18T11:31:38Z: Allocation "21950f1a" created: node "cab61feb", group "jaeger"
    2022-01-18T11:31:38Z: Allocation "2ccec6c0" created: node "cab61feb", group "zipkin"
    2022-01-18T11:31:38Z: Allocation "2d2808d9" created: node "cab61feb", group "prometheus"
    2022-01-18T11:31:38Z: Allocation "8416d461" created: node "cab61feb", group "otel-collector"
    2022-01-18T11:31:38Z: Allocation "16643334" created: node "cab61feb", group "otel-agent"
==> 2022-01-18T11:31:39Z: Monitoring evaluation "1f5b0e32"
    2022-01-18T11:31:39Z: Evaluation within deployment: "4d53f739"
    2022-01-18T11:31:39Z: Evaluation status changed: "pending" -> "complete"
==> 2022-01-18T11:31:39Z: Evaluation "1f5b0e32" finished with status "complete"
==> 2022-01-18T11:31:39Z: Monitoring deployment "4d53f739"
  ! Deployment "4d53f739" failed

    2022-01-18T11:41:38Z
    ID          = 4d53f739
    Job ID      = otel-demo
    Job Version = 0
    Status      = failed
    Description = Failed due to progress deadline

    Deployed
    Task Group       Desired  Placed  Healthy  Unhealthy  Progress Deadline
    jaeger           1        1       1        0          2022-01-18T11:42:02Z
    load-generators  1        1       1        0          2022-01-18T11:42:03Z
    otel-agent       1        4       0        4          2022-01-18T11:41:38Z
    otel-collector   1        4       0        4          2022-01-18T11:41:38Z
    prometheus       1        1       1        0          2022-01-18T11:42:08Z
    zipkin           1        1       1        0          2022-01-18T11:42:23Z
```
It would be helpful if you could share any pointers in resolving the above-mentioned issue.

**Comment by lgfa29** [comment]
Thanks for the report @odidev. It seems like I missed some of the latest updates to the Otel Collector. I've updated the examples to match the new [Kubernetes files](https://raw.githubusercontent.com/open-telemetry/opentelemetry-collector/main/examples/k8s/otel-config.yaml).

Could you try again and see if it's working now?

Thanks!

**Comment by odidev** [comment]
Hi @lgfa29 

Thanks for the response, as per your suggestion I checked by running open-telemetry in different deployment scenarios again on amd64 and arm64 platform, please find the result below: 

**AMD64:**  

As mentioned in the previous comment Gateway and demo were failing with ``` progress deadline``` issue but with the latest code both are getting deployed Successfully. 

**ARM64:** 

Earlier I was getting the same result on arm64 platform as well i.e. failure with ``` progress deadline``` issue although with the latest code too I am getting the same failure, please find the logs below: 

1. **Gateway** 

```
root@master-node:/home/ubuntu/nomad-open-telemetry-getting-started# nomad run examples/nomad/otel-collector.nomad
==> 2022-02-07T11:25:30Z: Monitoring evaluation "4deb3ba0"
    2022-02-07T11:25:30Z: Evaluation triggered by job "otel-collector"
==> 2022-02-07T11:25:31Z: Monitoring evaluation "4deb3ba0"
    2022-02-07T11:25:31Z: Evaluation within deployment: "0ed7b55c"
    2022-02-07T11:25:31Z: Allocation "36acf13b" created: node "ceb3c773", group "otel-collector"
    2022-02-07T11:25:31Z: Evaluation status changed: "pending" -> "complete"
==> 2022-02-07T11:25:31Z: Evaluation "4deb3ba0" finished with status "complete"
==> 2022-02-07T11:25:31Z: Monitoring deployment "0ed7b55c"
  ! Deployment "0ed7b55c" failed

    2022-02-07T11:35:30Z
    ID          = 0ed7b55c
    Job ID      = otel-collector
    Job Version = 0
    Status      = failed
    Description = Failed due to progress deadline

    Deployed
    Task Group      Desired  Placed  Healthy  Unhealthy  Progress Deadline
    otel-collector  1        5       0        4          2022-02-07T11:35:30Z

```

2. **Demo**

```
root@master-node:/home/ubuntu/sakshi/nomad-open-telemetry-getting-started# nomad run examples/nomad/otel-demo.nomad
==> 2022-02-07T11:36:39Z: Monitoring evaluation "cf4d6e6b"
    2022-02-07T11:36:39Z: Evaluation triggered by job "otel-demo"
    2022-02-07T11:36:39Z: Allocation "606d870c" created: node "ceb3c773", group "jaeger"
    2022-02-07T11:36:39Z: Allocation "69982d5c" created: node "ceb3c773", group "otel-collector"
    2022-02-07T11:36:39Z: Allocation "70e68c0e" created: node "ceb3c773", group "prometheus"
    2022-02-07T11:36:39Z: Allocation "745766c2" created: node "ceb3c773", group "load-generators"
    2022-02-07T11:36:39Z: Allocation "92eca6a6" created: node "ceb3c773", group "zipkin"
    2022-02-07T11:36:39Z: Allocation "d5d7d62e" created: node "ceb3c773", group "otel-agent"
==> 2022-02-07T11:36:40Z: Monitoring evaluation "cf4d6e6b"
    2022-02-07T11:36:40Z: Evaluation within deployment: "2314423b"
    2022-02-07T11:36:40Z: Evaluation status changed: "pending" -> "complete"
==> 2022-02-07T11:36:40Z: Evaluation "cf4d6e6b" finished with status "complete"
==> 2022-02-07T11:36:40Z: Monitoring deployment "2314423b"
  ! Deployment "2314423b" failed

    2022-02-07T11:46:39Z
    ID          = 2314423b
    Job ID      = otel-demo
    Job Version = 0
    Status      = failed
    Description = Failed due to progress deadline

    Deployed
    Task Group       Desired  Placed  Healthy  Unhealthy  Progress Deadline
    jaeger           1        1       1        0          2022-02-07T11:47:00Z
    load-generators  1        1       1        0          2022-02-07T11:47:00Z
    otel-agent       1        5       0        4          2022-02-07T11:46:39Z
    otel-collector   1        5       0        4          2022-02-07T11:46:39Z
    prometheus       1        1       1        0          2022-02-07T11:47:01Z
    zipkin           1        3       0        2          2022-02-07T11:46:39Z
```

It would be helpful if you could share any pointers on issues in arm64 platform.

**Comment by lgfa29** [comment]
I'm pretty sure the issue in ARM is that the image used for the demo is only built for AMD64:
https://hub.docker.com/layers/otel/opentelemetry-collector/0.38.0/images/sha256-940ede6cedd3b812097d451854e993f1366d2992354e7705c34da3079df20573?context=explore

You can double check this by looking at the allocation status for those failing groups using the [`nomad alloc status`](https://www.nomadproject.io/docs/commands/alloc/status) command.

From the `Gateway` output you sent, you can get the allocation ID and run, for example,  `nomad alloc status 36acf13b` (the ID will be different if you run it again), or you can check the status in the Nomad UI by visiting http://localhost:4646.

The images `0.43.0` and `0.44.0` do have ARM builds, but I don't know if the configuration will still work on those versions, and I'm trying to keep this repo in-sync with the Kubernetes demo.

But you can try updating the image yourself by changing the `otel_image` variable at the top of the sample job files. If you still see an error check the `nomad alloc status <ALLOC ID>` command as sent above and also check the application log using the [`nomad alloc logs`](https://www.nomadproject.io/docs/commands/alloc/logs) command.

**Comment by lgfa29** [comment]
Fixed in #5
